.github/scripts/promote_branch.sh

```diff
@@ -0,0 +1,160 @@
+    echo
+    echo "  --target-branch has to be specified."
+}
+
+check_if_branch_differs() {
+    ACTUAL_DIFFERENT_LINES=$(git diff --numstat origin/$1 | wc -l)
+    if [ $ACTUAL_DIFFERENT_LINES -ne 0 ] ; then
+        echo "Lines differ in branch $1"
+        echo "Actual differing lines: $(git diff --numstat origin/$1)"
+        exit 1
+    fi
+}
+
+check_if_any_commits_in_last_week() {
+    NEW_COMMITS=$(git log --oneline --since="$(date --date="7 days ago" +%Y-%m-%d)" | wc -l)
+    if [ $NEW_COMMITS -ne 0 ] ; then
+        echo "There are commits in staging that are less than a week old. Blocking promotion to promotion"
+        echo "Commits less than a week old: $(git log --oneline --since="$(date --date="7 days ago" +%Y-%m-%d)")"
+        exit 1
+    fi
+}
+
+if [ -z "${TARGET_BRANCH}" ]; then
+    echo -e "Error: missing 'target-branch' argument\n"
+    print_help
+    exit 1
+fi
+if [ "${TARGET_BRANCH}" == "staging" ]; then
+    SOURCE_BRANCH="development"
+elif [ "${TARGET_BRANCH}" == "production" ]; then
+    SOURCE_BRANCH="staging"
+else
+    echo "Invalid target-branch. Only 'staging' and 'production' are allowed"
+    print_help
+    exit 1
+fi
+if [ -z "${GITHUB_TOKEN}" ]; then
+    echo -e "Error: missing 'GITHUB_TOKEN' environment variable\n"
+    print_help
+    exit 1
+fi
+
+# Personal access token with appropriate permissions
+token="${GITHUB_TOKEN}"
+
+# Clone the repository
+tmpDir=$(mktemp -d)
+releaseServiceCatalogDir=${tmpDir}/release-service-catalog
+mkdir -p ${releaseServiceCatalogDir}
+
+echo -e "---\nPromoting release-service-catalog ${SOURCE_BRANCH} to ${TARGET_BRANCH}\n---\n"
+
+git clone "git@github.com:$ORG/$REPO.git" ${releaseServiceCatalogDir}
+cd ${releaseServiceCatalogDir}
+
+# A change cannot go into production if the changes in staging are less than a week old
+if [[ "${TARGET_BRANCH}" == "production" && "${OVERRIDE}" != "true" ]] ; then
+    git checkout origin/staging
+    check_if_any_commits_in_last_week
+fi
+
+# A change cannot go into staging if staging and production differ
+if [[ "${TARGET_BRANCH}" == "staging" && "${FORCE_TO_STAGING}" != "true" ]] ; then
+    git checkout origin/staging
+    update_taskGitRevision_in_pipelines production
+    check_if_branch_differs production
+fi
+
+description="Included PRs:\r\n"
+COMMITS=($(git rev-list --first-parent --ancestry-path origin/"$TARGET_BRANCH"'...'origin/"$SOURCE_BRANCH"))
+## now loop through the above array
+for COMMIT in "${COMMITS[@]}"
+do
+  PR_URL=$(curl -s   -H 'Authorization: token  '"$token"  'https://api.github.com/search/issues?q=sha:'"$COMMIT" | jq -r '.items[]
+    | select(.repository_url=="https://api.github.com/repos/'"$ORG"'/'"$REPO"'")
+    | .pull_request | select(.merged_at!=null) | .html_url')
+  description="$description"' - '"$PR_URL"'\r\n'
+done
+echo $description
+
+if [ "${DEBUG}" == "true" ] ; then
+    exit
+fi
+
+git checkout $SOURCE_BRANCH
+git checkout -b $TARGET_BRANCH
+git push origin $TARGET_BRANCH
```

Comment:
We can also just do `git push origin $SOURCE_BRANCH:$TARGET_BRANCH` and remove the `checkout -b` above. Since it's all deleted next, there is probably no need for the branch. But it's a minor thing.

Reply:
Nice, didn't know about that syntax. Updated

Reply:
Yeah, I remember this because it's basically the only way to delete a remote branch - to push "nothing" to it with `git push origin :$BRANCH`